runtime/wasm/src/asc_abi/class.rs

```diff
@@ -17,29 +18,52 @@ pub(crate) struct ArrayBuffer<T> {
     // elements in `content` will be aligned for any element type.
     _padding: [u8; 4],
```

Comment:
Unrelated to this PR, but do you know if this padding is actually needed? If we're transmuting ArrayBuffer, I don't think this is safe because Rust's memory layout for structs is undefined IIRC (except with `repr`)

Reply:
It's not needed, it's only there to illustrate the layout. We don't transmute it because ArrayBuffer has a custom impl of AscType. I've refactored so we use that field in `to_asc_bytes`.